components/script/dom/htmlimageelement.rs

```diff
@@ -782,6 +798,63 @@ impl LayoutHTMLImageElementHelpers for LayoutJS<HTMLImageElement> {
     }
 }
 
+pub fn parse_a_sizes_attribute(input: DOMString, width: Option<u32>) -> Vec<Size> {
```

Comment:
Add a comment like `// https://html.spec.whatwg.org/multipage/images.html#parse-a-sizes-attribute` so it's clear which algorithm this is based on.